src/gdb_hostio.c

```diff
@@ -24,24 +24,26 @@
 #include "gdb_hostio.h"
 #include "gdb_packet.h"
 
+#if PC_HOSTED == 0
```

Comment:
Just realised re-reading this part of the patch that this means the whole file is gated behind this check if we're building BMDA or not - perhaps rather than topping and tailing the file, we can modify the build system instead to omit this file under BMDA?

Reply:
I decided to drop that commit and leave it be in favor of some dedicated PR. The 7 warnings for `hosted` disappear anyway after brushing up gdb_hostio.c for PRIX32.